### ISSUE TITLE
3.2 Release Tracking

### ISSUE THREAD

**Issue by goanpeca** [jupyterlab/jupyterlab#10726]
This issue will gather all issues we will be working on for the 3.2 release.

The list will be updated with issues and PRs that will be included.

This release will have a span of 1-2 months so he initial 3.2rc will be planned for the end of September.

## Work
-  Toolbar customization [#10469](https://github.com/jupyterlab/jupyterlab/pull/10469)
-  Rich variable rendering [#10299](https://github.com/jupyterlab/jupyterlab/pull/10299) and restart and debug [#10462](https://github.com/jupyterlab/jupyterlab/pull/10462) PRs would be nice to get in 3.2
- Set `numberCellsToRenderDirectly` to `Inf` for 3.1 and enable with lower number in 3.2

**Comment by krassowski** [comment]
- https://github.com/jupyterlab/jupyterlab/pull/10632 (to be discussed)

**Comment by joelostblom** [comment]
- Would there be room for https://github.com/jupyterlab/jupyterlab/pull/9068 be considered for 3.2?

**Comment by goanpeca** [comment]
Hi @joelostblom that is rather a large endeavor that has stalled, so unless someone champions it full time, I do not think it will make it.

**Comment by mlucool** [comment]
Can we add fixes to the document undo/redo https://github.com/jupyterlab/jupyterlab/issues/10694 (or maybe this should be 3.1.x)?

**Comment by krassowski** [comment]
The PR fixing #10694 is already tagged to 3.1.x (assuming we go with this fix). I would also champion for a fix to #10732 to go into 3.1.x.

**Comment by mlucool** [comment]
I only see a PR for undo/redo of outputs. Is there another PR for "allowing users to disable the document-wide history tracking when not in RTC mode"?

I agree that  #10732 should also go into 3.1.x.

**Comment by goanpeca** [comment]
>I only see a PR for undo/redo of outputs. Is there another PR for "allowing users to disable the document-wide history tracking when not in RTC mode"?

I am not sure we are tracking that as a separate issue. @krassowski do we have such issue?

**Comment by krassowski** [comment]
We didn't but now we do: #10791.

**Comment by echarles** [comment]
We will need also https://github.com/jupyterlab/jupyterlab/issues/10714 in 3.2

**Comment by krassowski** [comment]
Did your #10793 fix that? If yes, I would say we should backport to 3.1.

**Comment by echarles** [comment]
> Did your #10793 fix that? If yes, I would say we should backport to 3.1.

Yes, https://github.com/jupyterlab/jupyterlab/issues/10793  is a fix for https://github.com/jupyterlab/jupyterlab/issues/10714 in master and also ported and merged in 3.1.x with https://github.com/jupyterlab/jupyterlab/pull/10797

**Comment by krassowski** [comment]
Thanks! I closed that issue for now to make it clear. I think that 3.2 will just branch off 3.1 so we should not worry about picking up individual PRs that were already backported, but a confirmation from @goanpeca would help here.

**Comment by blink1073** [comment]
I've created the `3.2.x` branch and made an initial alpha release.  The PRs below still need manual backports (plus the follow up Galata PRs.  I'd like to finish backports and cut a beta early next week.  We have three open PRs that are targeting 3.2.x as well, I've pinged the authors on all of them.

https://github.com/jupyterlab/jupyterlab/pull/10462
https://github.com/jupyterlab/jupyterlab/pull/10469
https://github.com/jupyterlab/jupyterlab/pull/10632
https://github.com/jupyterlab/jupyterlab/pull/10829
https://github.com/jupyterlab/jupyterlab/pull/10904
https://github.com/jupyterlab/jupyterlab/pull/10989
https://github.com/jupyterlab/jupyterlab/pull/11021

**Comment by blink1073** [comment]
I updated the top level comment and milestone to reflect the new timeline discussed in today's dev meeting.

**Comment by blink1073** [comment]
The only remaining outstanding items for the RC release next Tuesday (10/5) are:

- [ ] https://github.com/jupyterlab/jupyterlab/pull/10469
- [ ] https://github.com/jupyterlab/jupyterlab/pull/10949

**Comment by echarles** [comment]
> Enable document wide history tracking #10949

https://github.com/jupyterlab/jupyterlab/pull/10949 can be remove from 3.2 as it depends on https://github.com/jupyterlab/jupyterlab/issues/11035

**Comment by blink1073** [comment]
Thanks @echarles, I meant to put #10769, updated.

**Comment by blink1073** [comment]
We had some issues related to webpack #11218 and #11234 that slowed us down today, I'll make the RC tomorrow.

**Comment by blink1073** [comment]
RC is out!  https://github.com/jupyterlab/jupyterlab/releases/tag/v3.2.0rc0
